general/development/policies/codingstyle/index.md

```diff
@@ -985,6 +985,113 @@ $string['overduehandlingautoabandon'] = 'Unfinished attempts are immediately dis
 
 Language strings should not contain or even rely on any leading or trailing whitespace. Such strings are not easy to be translated in the translation tool, AMOS.
 
+## Objects
+
+### Importing `stdClass`
+
+For namespaced classes, importing `stdClass` via the `use` operator or using its fully qualified name are both acceptable.
+
+<ValidExample>
+
+```php
+use stdClass;
+
+// Using stdClass via import.
+$foo = new stdClass();
+```
+
+```php
+// Using stdClass with its fully qualified name.
+$foo = new \stdClass();
+```
+
+</ValidExample>
+
+### Dynamic properties
+
+`stdClass` objects can be created with dynamic properties by casting an associative array to object or by instantiating a new `stdClass` object and assigning properties to it.
+
+<ValidExample>
+
+```php
+// Object casting.
+$foo = (object) [
+    'bar' => 1,
+    'baz' => 2,
+];
+
+// Setting dynamic properties.
+$foo = new stdClass();
+$foo->bar = 1;
+$foo->baz = 2;
+```
+
+</ValidExample>
+
+### Empty objects
+
+Generic PHP objects can be instantiated by the `new` operator or created by typecasting an empty array to object. Both can be used in a number of situations.
+
+<ValidExample>
+
+Empty object creation
+
+```php
+// Via the `new` operator.
+$foo = new \stdClass();
+
+// Via empty array object casting.
+$foo = (object) [];
+```
+
+Empty objects within an object with properties or arrays
+
+```php
+// Empty object within an object.
+$foo = new stdClass();
+$foo->bar = (object) [];
+
+// Empty object within an array.
+$boo = [
+    'far' => (object) [],
+];
+```
+
+```php
```

Comment:
I'd suggest to put this examples before the type-casting one. That way all the examples first have the code with the `stdClass` and later the `(object) []` one.

Or the opposite, move the other examples so we always have the typecasting first and the stdClass later, I don't mind. Jut all the examples in the very same order.

Reply:
Thanks, Eloy! I amended the patch so that the `(object) []` ones come first and they'll be consistent with the rest of the examples.